core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/workflow/add/ValidateAddRequestActivity.java

```diff
@@ -132,12 +139,11 @@ protected Sku determineSku(Product product, Long skuId, Map<String, String> attr
         if (sku == null && product != null) {
             // Set to the default sku
             if (cannotSellDefaultSku(product)) {
-                throw new RequiredAttributeNotProvidedException("Unable to find non-default sku matching given options and cannot sell default sku", null);
+                throw new InventoryUnavailableException("Sku is not Available");
```

Comment:
Yeah don't change this. RequiredAttributeNotProvidedException is exactly what needs to be thrown here. It indicates that the you didn't send valid product option values